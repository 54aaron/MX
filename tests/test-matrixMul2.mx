int main(){
    Matrix d;
    Matrix n;
    Matrix a;
    Matrix q;
    Matrix r;
    Matrix s;
    n = [[1,2,3],[4,5,6], [7,8,9]];
    a = [[9,8,7], [6,5,4], [3,2,1]];
    d = a *. n;
    q = [[1,2,3],[4,5,6]];
    r = [[9,8,7], [6,5,4], [3,2,1]];
    s = q *. r;
    prints("==============================================");
    prints("Matrix a = ");
    print_matrix(a);
    prints("==============================================");
    prints("Matrix n = ");
    print_matrix(n);
    prints("==============================================");
    prints("a *. n = ");
    print_matrix(d);
    prints("==============================================");
    prints("Matrix q = ");
    print_matrix(q);
    prints("==============================================");
    prints("Matrix r = ");
    print_matrix(r);
    prints("==============================================");
    prints("q *. r = ");
    print_matrix(s);
    prints("==============================================");
}
