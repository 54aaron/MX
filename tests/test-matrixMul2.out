==============================================
Matrix a = 
[ 9, 8, 7 ]
[ 6, 5, 4 ]
[ 3, 2, 1 ]

==============================================
Matrix n = 
[ 1, 2, 3 ]
[ 4, 5, 6 ]
[ 7, 8, 9 ]

==============================================
a *. n = 
[ 90, 114, 138 ]
[ 54, 69, 84 ]
[ 18, 24, 30 ]

==============================================
Matrix q = 
[ 1, 2, 3 ]
[ 4, 5, 6 ]

==============================================
Matrix r = 
[ 9, 8, 7 ]
[ 6, 5, 4 ]
[ 3, 2, 1 ]

==============================================
q *. r = 
[ 30, 24, 18 ]
[ 84, 69, 54 ]

==============================================
